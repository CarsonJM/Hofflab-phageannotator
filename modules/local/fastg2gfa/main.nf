process FASTG2GFA {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/phables:1.4.0--pyhdfd78af_0':
        'biocontainers/phables:1.4.0--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(fastg)

    output:
    tuple val(meta), path("${prefix}.gfa.gz")   , emit: gfa
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    ${projectDir}/bin/gfa1/misc/fastg2gfa \\
        ${fastg} > ${prefix}.gfa

    gzip ${prefix}.gfa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phables: \$(echo \$(phables -v 2>&1) | sed -n 's/phables, version //' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${prefix}.gfa.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phables: \$(echo \$(phables -v 2>&1) | sed -n 's/phables, version //' ))
    END_VERSIONS
    """
}
