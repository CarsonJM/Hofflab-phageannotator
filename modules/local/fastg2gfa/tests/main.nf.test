nextflow_process {

    name "Test Process FASTG2GFA"
    script "../main.nf"
    process "FASTG2GFA"

    test("fastg") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file("${projectDir}/assets/test-datasets/fastg/test1.graph.fastg.gz", checkIfExists: true) ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }


    test("fastg - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file("${projectDir}/assets/test-datasets/fastg/test1.graph.fastg.gz", checkIfExists: true) ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
