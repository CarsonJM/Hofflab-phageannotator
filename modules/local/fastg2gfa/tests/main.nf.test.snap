{
    "fastg": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gfa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f426919016db27fef3c3709f169df056"
                ],
                "gfa": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gfa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f426919016db27fef3c3709f169df056"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-23T14:21:59.13007263"
    },
    "fastg - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gfa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f426919016db27fef3c3709f169df056"
                ],
                "gfa": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gfa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f426919016db27fef3c3709f169df056"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-23T14:22:07.659145366"
    }
}